### DISCUSSION TITLE
#7 7th Sprint - ASR, API and login view

### DISCUSSION THREAD

**Discussion by Annalytic-programming** [CUMGroup/AMOGUS#39] [Blog]
Howdy AMOGUS friends ඞ,

this week we dealt with the [ASR](https://github.com/CUMGroup/AMOGUS/blob/main/documents/architecture/ASR-AMOGUS.md) we discussed in software engineering. 
We came to the conclusion that we want to implement our previously planned [ABBA-Scheme](https://github.com/CUMGroup/AMOGUS/blob/main/documents/architecture/BackendArchitecture.pdf) in this way. In addition, we continue to rely on a service-based architecture. Our frontend backend communication will be done via REST-API. To simplify deployment, we use separate docker containers for the frontend and backend. Thanks for your job on the ASR @jas20202 

In addition, we now have two new services in the frontend, which of course we don't want to withhold from you.
First, there is our API service that handles communication with the backend.
The second service is the authorization service which handles the login and registration for the client. These two were implemented by my humble self (@Annalytic-programming)

Furthermore we implemented an very important Service in our Backend. The exercise service. This resulted in a change that also affects our database schema.
We have decided to no longer store the complex tasks in the database, but record them as JSON objects which can easily be modified via your favorite editor (e.g. vim 😏)
Thanks for the efforts to our architecture and backend genius @AlexMi-Ha 

In order to be able to provide you with visible results, we have already dealt with the login/register view. Here is a screenshot of our preliminary prototype:

![grafik](https://user-images.githubusercontent.com/97878953/204585616-f07e86b4-3e7a-429f-a17e-788c630a965d.png)
![grafik](https://user-images.githubusercontent.com/97878953/204585649-94af60bc-5505-4f2f-ab47-e6094a49fb00.png)
Thanks to our "nearly" senior frontend dev @WachtelHD ^^

Next in the pipeline is our profile view. We hope to give you a glimpse of it next week.
Thanks for your work so far @elnaz-gharoon 

Yours truly,
CUMGroup

P.S. stay sus 🍌
![ezgif-3-b927506b5e](https://user-images.githubusercontent.com/97878953/204599762-d342b183-e457-4b44-8039-b9d26cfcbbf6.gif)

**Comment by EhrlerL** [comment]
\*jumps\* Hello there!
While I was reading your blog and ASR, I had a calming feeling of familiarity. This is because you have chosen the right - I mean you have chosen a good architechture (the same as we have). For us, it also felt like the best approach to split the project into several components and micro services. 
Since our utility tree is focused on availability, I don't have a direct comparison to yours but it seems you have well-chosen requirements. 
Your new demo looks very "schnieke" (already excited about the next one). Imo the register/login-switch is a bit uncommon and confusing at first. I would rather choose the "classic" way using a lil text like "Already have an account? Login [here](http://amogus.org/).
Good luck on the tasks for next week.
Cheers, capyclue. ₍ᐢ•(ܫ)•ᐢ₎

**Reply by Annalytic-programming** [reply]
Heyyoo capyclue,
thanks for your suggestion for the register/login-switch. We already discussed this in our team. We decided to change that by time since the views so far are prototypes and could reworked easily.
Futhermore we want to thank you for your nice comment ^^
Yours truly
CUMGroup

**Comment by entrisity** [comment]
Hello there, 
the software architecture that you guys have chosen seems very promising. Your diagram for your ABBA-Scheme also looks fine to me. Regarding the login view make sure you keep your capitilization consistent and have even spacing between components to further maintain consistency. Excited to take a look at your next views that you will be implementing in the next few weeks, especially the profile view!

-Team cybuy

**Reply by jas20202** [reply]
Hello there team CyBuy, 
better later than never, here is a response to your comment:
Thank you very much for looking through our software architecture ^^ As for our login page: it's just a demo so far and of course we will make the capitalization consistent in the end. Unfortunately our demo for the profile view is not done yet and we hope to have it by the beginning of next semester. 

We wish you a happy new Year and a great practical phase \\(^-^)/

Yours truly,
CUMGroup (Team AMOGUS)

**Comment by Richone01** [comment]
Hi!
I really like your software architecture style and I think it is going to fit your project very well.  We chose the same one and facing deployment and updates, a service based architecture with different docker containers makes your life much easier. I also like the design of your first views. It looks quite simple but yet very nice an d smooth. Really like the idea with the register/login slider!
I hope you keep up the great work!

~Team DHBWorkout

**Reply by jas20202** [reply]
Hey hey DHBWorkout,

thanks for dropping in and looking over our updates. Good to know that your team chose the same architecture styles so we could mayb exchange our experiences with it. It's nice to hear that you like our simple designs! We are looking forward to keep it simple and consistent.

Until then we wish you a happy new Year and a great practical phase \(^-^)/

Yours truly,
CUMGroup (Team AMOGUS)
